### ISSUE TITLE
Consider enabling Clinic analysis to simplify logical operations (LogicalAnd, LogicalOr)

### ISSUE THREAD

**Issue by bluesadi** [angr/angr#3784]
### Description

C source code:
```cpp
int32_t land32(int32_t a, int32_t b){
    return a && b;
}
```
angr-management's decompilation result versus IDA Pro's:
![image](https://user-images.githubusercontent.com/35004522/216754891-6562984c-bf92-4a2f-b08f-b60f1ea1fdf0.png)
![image](https://user-images.githubusercontent.com/35004522/216754904-69d1b655-3b54-4d55-85d6-e90977295416.png)
ARM/MIPS/X86 binaries all have this problem. I did not test decompiling LogicalOr, but I suppose they are alike.

I believe simplifying that if-else structure to some simpler one (say LogicalAnd expression) would facilitate both reverse engineering and static analysis.

[land32.zip](https://github.com/angr/angr/files/10608153/land32.zip)
[land32_arm.zip](https://github.com/angr/angr/files/10608154/land32_arm.zip)
[land32_mips.zip](https://github.com/angr/angr/files/10608155/land32_mips.zip)

**Comment by ltfish** [comment]
Try switching the structuring algorithm to Phoenix.

**Comment by bluesadi** [comment]
There's no much difference

![P`BAQAUPIGMA)KYE 1WAXYQ](https://user-images.githubusercontent.com/35004522/216756609-bb3af278-bc05-4b1a-9935-2267be7b8a2c.png)

![RR7HYP{7_4}5`7)D8AXR$MW](https://user-images.githubusercontent.com/35004522/216756624-7bfc9812-bd1f-4f6d-a451-5bea89a153ad.png)

**Comment by ltfish** [comment]
Then turn off EagerReturnSimplifier?

**Comment by bluesadi** [comment]
![image](https://user-images.githubusercontent.com/35004522/216756953-3e13317c-5b07-4bfe-baa0-ffa0323ea431.png)

**Comment by ltfish** [comment]
Yep. So with `max_level` set to 1 (by modifying `eager_return_simplifier.py`), you should see the expected result.

We have solved this problem in a development branch. It will be merged to master once the paper is accepted.

**Comment by bluesadi** [comment]
Cool! Wish your paper get accepted soon.

I've changed the parameter `max_level` to 1 at https://github.com/angr/angr/blob/02279878d30ac90301e9ff58841108b892fdc221/angr/analyses/decompiler/optimization_passes/eager_returns.py#L58

but it still did not yield the simplest code (`a && b`):

![image](https://user-images.githubusercontent.com/35004522/216760886-7afdcaa4-80e5-4d09-8d17-3b8ee658e788.png)

Are there any other changes needed?
